src/utils/propUtils.js

```diff
@@ -60,3 +63,15 @@ export const customPropTypes = {
   },
 }
 
+// ------------------------------------
+// Prop Renderers
+// ------------------------------------
+export const iconPropRenderer = (val) => _.isString(val) ? <Icon className={val} /> : val
+export const imagePropRenderer = (val) => _.isString(val) ? <Image src={val} /> : val
+
+// ------------------------------------
+// Prop To Class Name
+// ------------------------------------
+export const useKeyOnly = (val, key) => val && key
+export const useValueAndKey = (val, key) => val && val !== true && `${val} ${key}`
+export const useKeyOrValueAndKey = (val, key) => val && (val === true ? key : `${val} ${key}`)
```

Comment:
There are 4 prop patterns used to build up the `className` for a component.  The one not shown above is `valueOnly()` because it would simply return `val`, so the value is used inline instead.

**Key Only**
Props where the name only is used.

`<div class="ui tag label"></div>`
`<Label tag />`

**Value Only**
Props where only the value is used.

`<div class="ui big label"></div>`
`<Label size='big' />`

**Value And Key**
Props that require a value, where className requires value and key.

`<div class="ui left corner label"></div>`
`<Label corner='left' />`

**Key || Value And Key**
Boolean props with optional value, where className includes the key or value and key.

`<div class="ui pointing label"></div>`
`<Label pointing />`

`<div class="ui left pointing label"></div>`
`<Label pointing='left' />`